### PR TITLE
Fix: coreos-livepxe-rootfs to print rootfs_url

### PR DESCRIPTION
**Pull request by jmarrero**

### CODE REVIEW

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ elif [[ -n "${rootfs_url}" ]]; then
     if [[ ${rootfs_url} != http:* && ${rootfs_url} != https:* ]]; then
         # Don't commit to supporting protocols we might not want to expose in
         # the long term.
-        echo "coreos.live.rootfs_url= supports HTTP and HTTPS only." >&2
+        echo "coreos.live.rootfs_url=${rootfs_url} supports HTTP and HTTPS only." >&2
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: Maybe here let's reword as e.g.

```suggestion
        echo "coreos.live.rootfs_url=${rootfs_url} uses unsupported scheme. Only HTTP and HTTPS are supported." >&2
```

?

**Review by jlebon — DISMISSED** [review]
One suggestion, otherwise LGTM.

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh (hunk 1 above)
Comment (jmarrero) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: I like this. Adding it.

Comment (jmarrero) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: I made it: `uses an unsupported scheme` WDYT?

**Comment by bgilbert** [comment]
Thanks for taking this!

It's worth considering how the line wrapping will look on an actual 80-character-wide console, particularly with arbitrarily-long URLs.  It may be worth putting the karg on its own line.

**Comment by jmarrero** [comment]
> Thanks for taking this!
> 
> It's worth considering how the line wrapping will look on an actual 80-character-wide console, particularly with arbitrarily-long URLs. It may be worth putting the karg on its own line.

updated in latest push.

**Comment by jmarrero** [comment]
Testing error message change was successful: 
```
[  OK  ] Finished dracut pre-mount hook.
[    5.303124] audit: type=1130 audit(1635095549.195:26): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-pre-mount comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[    5.304693] systemd[1]: Finished dracut pre-mount hook.
[    5.308584] coreos-livepxe-rootfs[571]: Fetching rootfs image from http://tesdkljhflskdhjf.1ggd8.120.1:8000/kickstart.cfg...
[    5.315137] coreos-livepxe-rootfs[591]: curl: (6) Could not resolve host: tesdkljhflskdhjf.1ggd8.120.1
[    5.316387] coreos-livepxe-rootfs[571]: Couldn't establish connectivity with the server specified by:
[    5.317131] coreos-livepxe-rootfs[571]: coreos.live.rootfs_url=http://tesdkljhflskdhjf.1ggd8.120.1:8000/kickstart.cfg
[    5.317899] coreos-livepxe-rootfs[571]: Retrying in 5s...   
```

```
-- Journal begins at Sun 2021-10-24 17:15:02 UTC, ends at Sun 2021-10-24 17:15:06 UTC. --
Oct 24 17:15:06 systemd[1]: Starting Acquire live PXE rootfs image...
Oct 24 17:15:06 coreos-livepxe-rootfs[569]: Fetching rootfs image from git://192.168.120.1:8000/kickstart.cfg...
Oct 24 17:15:06 coreos-livepxe-rootfs[569]: Only HTTP and HTTPS are supported. Currently the image is specified by:
Oct 24 17:15:06 coreos-livepxe-rootfs[569]: coreos.live.rootfs_url=git://192.168.120.1:8000/kickstart.cfg
Oct 24 17:15:06 coreos-livepxe-rootfs[569]: Please fix your PXE configuration as it is set to use an unsupported scheme.
Oct 24 17:15:06 systemd[1]: coreos-livepxe-rootfs.service: Main process exited, code=exited, status=1/FAILURE
Oct 24 17:15:06 systemd[1]: coreos-livepxe-rootfs.service: Failed with result 'exit-code'.
```

```
Oct 24 17:16:53 coreos-livepxe-rootfs[592]: Error: hash mismatch at offset 0; expected 75bffca6f26345793f4f52d3b2c9db29e4b727f181f94dac4579a048f86e6ed7, found de6b604cd233a85de92f43fc79534d3e18aca1820ab6c6de11b8e3a740482468
Oct 24 17:16:53 coreos-livepxe-rootfs[570]: Couldn't fetch, verify, and unpack image specified by:
Oct 24 17:16:53 coreos-livepxe-rootfs[570]: coreos.live.rootfs_url=http://192.168.120.1:8000/kickstart.cfg
Oct 24 17:16:53 coreos-livepxe-rootfs[570]: Check that the URL is correct and that the rootfs version matches the initramfs.
Oct 24 17:16:53 systemd[1]: coreos-livepxe-rootfs.service: Main process exited, code=exited, status=1/FAILURE
```

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh [hunk 2]
```diff
@@ -25,9 +25,10 @@ elif [[ -n "${rootfs_url}" ]]; then
     echo "Fetching rootfs image from ${rootfs_url}..."
     if [[ ${rootfs_url} != http:* && ${rootfs_url} != https:* ]]; then
         # Don't commit to supporting protocols we might not want to expose in
-        # the long term.
-        echo "coreos.live.rootfs_url= supports HTTP and HTTPS only." >&2
-        echo "Please fix your PXE configuration." >&2
+        # the long term.
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: Minor: looks like trailing space was added here.

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh [hunk 3]
```diff
@@ -25,9 +25,10 @@ elif [[ -n "${rootfs_url}" ]]; then
     echo "Fetching rootfs image from ${rootfs_url}..."
     if [[ ${rootfs_url} != http:* && ${rootfs_url} != https:* ]]; then
         # Don't commit to supporting protocols we might not want to expose in
-        # the long term.
-        echo "coreos.live.rootfs_url= supports HTTP and HTTPS only." >&2
-        echo "Please fix your PXE configuration." >&2
+        # the long term.        
+        echo "Only HTTP and HTTPS are supported. Currently the image is specified by:" >&2
+        echo "coreos.live.rootfs_url=${rootfs_url}" >&2
+        echo "Please fix your PXE configuration as it is set to use an unsupported scheme." >&2
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: To follow the same format as the other errors, maybe:

```suggestion
        echo "Unsupported scheme for image specified by:" >&2
        echo "coreos.live.rootfs_url=${rootfs_url}" >&2
        echo "Only HTTP and HTTPS are supported. Please fix your PXE configuration." >&2
```

?

**Review by jlebon — DISMISSED** [review]
Some minor tweaks, but cool to do in a follow-up too!

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh (hunk 3 above)
Comment (jmarrero) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: Updated

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh (hunk 2 above)
Comment (jmarrero) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: removed

**Review by jlebon — APPROVED** [review]

**Comment by jmarrero** [comment]
@jlebon thanks!

**Comment by jlebon** [comment]
For reference, the RHBZ this pertains to is https://bugzilla.redhat.com/show_bug.cgi?id=2005507. (Ideally, should've been part of the commit message but missed it during review! :) )

**Comment by jmarrero** [comment]
> For reference, the RHBZ this pertains to is https://bugzilla.redhat.com/show_bug.cgi?id=2005507. (Ideally, should've been part of the commit message but missed it during review! :) )

Thanks will make sure to add it the next time.

**Pull request merged**